casper/src/main/scala/coop/rchain/casper/dag/BlockDagKeyValueStorage.scala

```diff
@@ -233,12 +243,68 @@ final class BlockDagKeyValueStorage[F[_]: Concurrent: Log] private (
   override def addDeploy(d: Signed[DeployData]): F[Unit] = deployStore.put(d.sig, d)
 
   override def pooledDeploys: F[Map[DeployId, Signed[DeployData]]] = deployStore.toMap
+
+  // Map of deploys being executed and execution results
+  private val execMap    = TrieMap.empty[DeployId, Option[String]]
+  private val expiredMap = TrieMap.empty[DeployId, Unit]
+
+  def deployStatus(d: DeployId): F[String] = blockMetadataIndex.finalizedBlockSet.flatMap {
```

Comment:
Could we return `F[someType]` instead of `F[String]`.

Reply:
This is a purpose of my next PR so don't worry about it.